src/lib/storage/table.cpp

```diff
@@ -1,89 +1,118 @@
 #include "table.hpp"
 
 #include "utils/assert.hpp"
+#include "value_segment.hpp"
 
 namespace opossum {
 
 Table::Table(const ChunkOffset target_chunk_size) {
-  // Implementation goes here
+  _target_chunk_size = target_chunk_size;
+  _chunks = std::vector<std::shared_ptr<Chunk>>();
+  _column_names = std::vector<std::string>();
+  _column_types = std::vector<std::string>();
+  _column_nullable = std::vector<bool>();
+  create_new_chunk();
 }
 
 void Table::add_column_definition(const std::string& name, const std::string& type, const bool nullable) {
-  // Implementation goes here
+  _column_names.emplace_back(name);
+  _column_types.emplace_back(type);
+  _column_nullable.emplace_back(nullable);
 }
 
 void Table::add_column(const std::string& name, const std::string& type, const bool nullable) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  for (auto chunk : _chunks) {
+    auto new_segment = std::shared_ptr<AbstractSegment>{};
+    if (type == "string") {
+      new_segment = std::make_shared<ValueSegment<std::string>>(nullable);
+    } else {
+      new_segment = std::make_shared<ValueSegment<int32_t>>(nullable);
+    }
+    chunk->add_segment(new_segment);
+  }
+  add_column_definition(name, type, nullable);
+  _column_count++;
 }
 
 void Table::create_new_chunk() {
-  // Implementation goes here
+  auto new_chunk = std::make_shared<Chunk>();
+  for (int index = 0; index < _column_count; ++index) {
```

Comment:
Consider using the appropriate data type here:
```suggestion
  for (auto index = uint16_t{0}; index < _column_count; ++index) {
```